app/questionnaire/value_source_resolver.py

```diff
@@ -272,6 +272,14 @@ def _resolve_summary_with_calculation(
             data_stores=self.data_stores,
             location=self.location,
             routing_path_block_ids=self.routing_path_block_ids,
+            value_source_resolver=ValueSourceResolver(
```

Comment:
We definitely don't want to instantiate another `ValueSourceResolver` inside the `ValueSourceResolver` class instance.

Reply:
I agree with @berroar this probably needs some more thought and some kind of structured approach/plan/proposal.

Reply:
Yeah this felt akward to do when refactoring, Will have a look